MH4U Dex/CoreDataController.m

```diff
@@ -363,6 +439,87 @@ - (NSUInteger)countForEntityWithEntityName:(NSString *)entityName withPredicate:
+        item.name = dropDict[QuestDropItemNameKey];
+    }
+    drop.item = item;
+    return drop;
+}
+
+- (Quest *)questFromDictionary:(NSDictionary *)questDict
+{
+    Quest *quest = [Quest insertInManagedObjectContext:self.managedObjectContext];
+    quest.id = questDict[QuestIDKey];
+    quest.name = questDict[QuestNameKey];
+    if ([questDict[QuestCaravanHallKey] isEqualToString:QuestCaravanKey]) {
+        quest.caravanValue = YES;
+    } else {
+        quest.caravanValue = NO;
+    }
+    if ([questDict[QuestDangerKey] isEqualToString:TrueString]) {
+        quest.dangerValue = YES;
+    } else {
+        quest.dangerValue = NO;
+    }
+    quest.fee = questDict[QuestFeeKey];
+    quest.hrp = questDict[QuestHRPKey];
+    if ([questDict[QuestKeyIndicatorKey] isEqualToString:TrueString]) {
+        quest.keyValue = YES;
+    } else {
+        quest.keyValue = NO;
+    }
+    quest.objective = questDict[QuestObjectiveKey];
+    quest.subObjective = questDict[QuestSubQuestObjectiveKey];
+    quest.reward = questDict[QuestRewardKey];
+    quest.stars = questDict[QuestStarsKey];
+    quest.type = questDict[QuestTypeKey];
+    if ([questDict[QuestUrgentKey] isEqualToString:TrueString]) {
+        quest.urgentValue = YES;
+    } else {
+        quest.urgentValue = NO;
+    }
+    //TODO: add in outbreak bool handling once that data can be verified.
+    
+    NSPredicate *regionPredicate = [NSPredicate predicateWithFormat:@"%K == %@", RegionAttributes.keyName, questDict[QuestMapKeyNameKey]];
+    // If this method call returns nil, then quest.region will be nil.  That is okay!
+    quest.region = (Region *)[self getUniqueEntityWithEntityName:[Region entityName] withPredicate:regionPredicate];
+    
+    NSPredicate *monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[QuestFirstMonsterKey]];
+    // The same is the case with the following monster-fetching calls
+    quest.firstMonster = (Monster *)[self getUniqueEntityWithEntityName:[Monster entityName] withPredicate:monsterPredicate];
+    if (quest.firstMonster) {
+        //Continue on to the second, and so on.  By nesting this way, we can avoid unnecessary fetches
+        monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[QuestSecondMonsterKey]];
+        quest.secondMonster = (Monster *)[self getUniqueEntityWithEntityName:[Monster entityName] withPredicate:monsterPredicate];
+        if (quest.secondMonster) {
+            monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[QuestThirdMonsterKey]];
+            quest.thirdMonster = (Monster *)[self getUniqueEntityWithEntityName:[Monster entityName] withPredicate:monsterPredicate];
+            if (quest.thirdMonster) {
+                monsterPredicate = [NSPredicate predicateWithFormat:@"%K == %@", MonsterAttributes.name, questDict[QuestFourthMonsterKey]];
```

Comment:
It may be worth creating a method on, say, the `Monster` class along the lines of `predicateForMonsterWithName:(NSString *)name` to make these predicate-creation lines shorter and less repetitive.